apps/bytebank-native/components/TransactionAttachment.tsx

```diff
@@ -0,0 +1,236 @@
+import * as Sharing from 'expo-sharing';
+
+import { AttachmentCard } from './AttachmentCard';
+import { FileUpload } from './FileUpload';
+import { DeleteAttachmentModal } from './DeleteAttachmentModal';
+import { TransactionAttachmentService } from '@/lib/services/attachment-service';
+import { TransactionsQueriesService } from '@fiap-tech-challenge/database/queries';
+import { supabase } from '@/lib/supabase';
+import { useAsyncAction } from '@/hooks/useAsyncOperation';
+
+export interface BaseTransaction {
+  id: string;
+  attachment_url?: string | null;
+  attachment_name?: string | null;
+}
+
+interface TransactionAttachmentProps {
+  transaction?: BaseTransaction;
+  onAttachmentChange?: (transaction: BaseTransaction) => void;
+  onFileSelect?: (file: { name: string; uri: string; type: string; size: number } | null) => void;
+  disabled?: boolean;
+  mode?: 'create' | 'edit';
+  className?: string;
+}
+
+export function TransactionAttachment({
+  transaction,
+  onAttachmentChange,
+  onFileSelect,
+  disabled = false,
+  mode = 'edit',
+  className,
+}: TransactionAttachmentProps) {
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<{ name: string; uri: string; type: string; size: number } | null>(null);
+  
+  const attachmentService = new TransactionAttachmentService();
+  const transactionService = new TransactionsQueriesService(supabase);
+
+  const uploadOperation = useAsyncAction({
+    onSuccess: () => Alert.alert('Sucesso', 'Arquivo anexado com sucesso!'),
+    onError: (error) => `Falha ao anexar arquivo: ${error instanceof Error ? error.message : 'Erro desconhecido'}`,
+  });
+
+  const downloadOperation = useAsyncAction({
+    onError: () => 'Não foi possível baixar o arquivo. Tente novamente.',
+  });
+
+  const deleteOperation = useAsyncAction({
+    onSuccess: () => Alert.alert('Sucesso', 'Anexo removido com sucesso!'),
+    onError: (error) => `Falha ao remover anexo: ${error instanceof Error ? error.message : 'Erro desconhecido'}`,
+  });
+
+  const isCreateMode = mode === 'create' || !transaction;
+  const hasAttachment = !isCreateMode && transaction?.attachment_name && transaction?.attachment_url;
+  const hasSelectedFile = isCreateMode && selectedFile;
+
+    const handleUpload = async (transactionId: string, file: { name: string; uri: string; type: string; size: number }) => {
+    try {
+      await attachmentService.uploadAttachment(transactionId, file);
+      
+      const updatedTransaction = await transactionService.getTransactionById(transactionId);
+      return {
+        url: updatedTransaction.attachment_url || '',
+        name: updatedTransaction.attachment_name || ''
+      };
+    } catch (error) {
+      console.error('Erro no upload:', error);
+      throw error;
+    }
+  };
+
+  const handleFileSelect = async (file: { name: string; uri: string; type: string; size: number } | null) => {
```

Comment:
Mais legível dessa forma:

```suggestion
const handleFileSelect = async (file: { name: string; uri: string; type: string; size: number } | null) => {
  if (!file) return;

  if (isCreateMode) {
    setSelectedFile(file);
    onFileSelect?.(file);
    return;
  }

  if (!transaction) return;

  await uploadOperation.execute(async () => {
    const { url, name } = await handleUpload(transaction.id, file);

    onAttachmentChange?.({
      ...transaction,
      attachment_url: url,
      attachment_name: name,
    });
  });
};

```